deps/verifier/src/snp/mod.rs

```diff
@@ -34,43 +35,43 @@ const LOADER_SPL_OID: Oid<'static> = oid!(1.3.6 .1 .4 .1 .3704 .1 .3 .1);
 
 #[derive(Debug)]
 pub struct Snp {
-    vendor_certs: VendorCertificates,
+    vendor_certs: Chain,
 }
 
-pub(crate) fn load_milan_cert_chain() -> &'static Result<VendorCertificates> {
-    static MILAN_CERT_CHAIN: OnceLock<Result<VendorCertificates>> = OnceLock::new();
-    MILAN_CERT_CHAIN.get_or_init(|| {
-        let certs = X509::stack_from_pem(include_bytes!("milan_ask_ark_asvk.pem"))?;
-        if certs.len() != 3 {
-            bail!("Malformed Milan ASK/ARK/ASVK");
+fn read_certs_from_file() -> Vec<X509> {
+    match X509::stack_from_pem(include_bytes!("milan_ask_ark_asvk.pem")) {
+        Result::Ok(certs) => {
+            if certs.len() != 3 {
+                panic!("Malformed Milan ASK/ARK/ASVK");
+            }
+            return certs;
         }
+        Result::Err(err) => panic!("Error reading certificates file: {err}"),
+    }
+}
 
-        let vendor_certs = VendorCertificates {
-            ask: certs[0].clone(),
-            ark: certs[1].clone(),
-            asvk: certs[2].clone(),
-        };
-        Ok(vendor_certs)
+pub(crate) fn load_milan_cert_chain() -> &'static Chain {
+    static MILAN_CERT_CHAIN: OnceLock<Chain> = OnceLock::new();
+    MILAN_CERT_CHAIN.get_or_init(|| {
+        let certs = read_certs_from_file();
+        Chain {
+            ca: CaChain {
+                ark: Certificate::from(certs[1].clone()),
+                ask: Certificate::from(certs[0].clone()),
+            },
+            vek: Certificate::from(certs[2].clone()),
```

Comment:
I thought this was the `asvk`. Has the term changed? If so, please remove references to `asvk`. I thought `vek` already referred to the guest encryption key.

Reply:
In this case, the `ASK` and the `VEK` are generic terms. The **A**MD **S**igning **K**ey (ASK) covers both the traditional `ASK` and the `ASVK`. The **V**ersioned **E**ndorsement **K**ey (VEK) covers `VCEK` or `VLEK`.

Reply:
Isn't `certs[2]` going to be the `ASVK`? That shouldn't go into the `VEK` field. The current `VendorCertificates` struct holds both the `ASK` and `ASVK`. It doesn't like the `CaChain` struct does that so you might need to tweak the logic a bit.

Reply:
You are absolutely correct. I misunderstood your comment. I will have to think about this, as a certificate chain usually should contain one or the other, not both.